client/src/Pages/playlistLength/PlaylistLength.jsx

```diff
@@ -1,46 +1,40 @@
 import React from "react";
-import { useContext, useState, useEffect } from "react";
+import { useContext, useState } from "react";
 import { UserContext } from "../../Context/UserContext";
-import { useNavigate } from "react-router-dom";
 import "./PlaylistLength.scss";
 import axios from "axios";
 import Slider from "@mui/material/Slider";
+import { useNavigate } from "react-router-dom";
 
 function PlaylistLength() {
   const [length, setLength] = useState(5);
   const user = useContext(UserContext);
-  // const BASE_URL = import.meta.env.VITE_BASE_URL;
-  const BASE_URL = "https://api.spotify.com/v1/";
-  let navigate = useNavigate();
+  const navigate = useNavigate();
+  // const BASE_URL = 'https://api.spotify.com/v1/';
 
   const getRecommendations = async (e) => {
     try {
-      console.log("getRecommendations", user.profile.access_token);
-      console.log("user artist", user.artists);
-      const { data } = await axios.get(
-        `${BASE_URL}recommendations?limit=${length}&seed_artists=${user.artists.join(
-          ","
-        )}&${user.mood}`,
-        {
-          headers: {
-            Authorization: `Bearer ${user.profile.access_token}`,
+      await axios
+        .post(
+          `http://localhost:8080/recommendations`,
+          {
+            length: length,
+            artists: user.artists.join(","),
+            mood: user.mood,
           },
-        }
-      );
-      console.log(data);
-      localStorage.setItem("recommended playlist", JSON.stringify(data));
-      user.setRecommendedData(data);
+          { withCredentials: true }
+        )
+        .then((res) => {
+          user.setRecommendedData(res.data.tracks);
```

Comment:
please change line 28 to:
`user.setRecommendedData(res.data);`

i just realized that the saved playlist needs an array instead of the 'tracks' object. Thanks

Reply:
@allansomera res.data returns both seeds & tracks, when I checked it with the playlist gen it seemed to be working fine since tracks is already an array of objects

Reply:
after testing a bit more, it looks like everything is working in order, not sure what caused the errors on my side. Resolving this convo